frontend/views/Login.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from "react";
+import { LinearGradient } from "expo-linear-gradient";
+import { Alert, View, StyleSheet } from "react-native";
+import { CommonActions } from "@react-navigation/native";
+import { Button, Card, TextInput } from "react-native-paper";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+
+import { Admin } from "../api/admin";
+import { updateUser } from "../redux/UserAction";
+
+const styles = StyleSheet.create({
+  card: {
+    marginTop: 100,
+    marginLeft: 20,
+    marginRight: 20,
+  },
+  container: {
+    padding: 10,
+    fontSize: 18,
+  },
+});
+
+function Login({ route, navigation, updateUser, user }) {
```

Comment:
For the props can you specify them in a similar fashion as `Culture`? That way we have type hints and all that.